newIDE/app/src/EventsSheet/index.js

```diff
@@ -363,6 +368,20 @@ export default class EventsSheet extends React.Component<Props, State> {
     return newEvents;
   };
 
+  //TODO ideally see for use toggle
+  openTextEditorDialog = () => {
```

Comment:
There you should probably get the selected event, store it in the state, and then use this state to pass it as a prop to the text editor dialog.

```
const selectedEvents = getSelectedEvents(this.state.selection);
if (!selectedEvents.length) return;
const event = selectedEvents[selectedEvents.length - 1]; // Get the last selected event.
this.setState({
  textEditedEvent: event;
});
```

You can see that a "bonus", we don't need the state "textEditorDialog". The fact that `textEditedEvent` is not null is enough for us to know that we must show the dialog :)

Reply:
This way of thinking is fantastic. I'll keep in my mind.
Beautiful idea!
Fixed